app/src/main/java/com/codesoom/assignment/controllers/SessionController.java

```diff
@@ -0,0 +1,45 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.application.AuthenticationService;
+import com.codesoom.assignment.application.UserService;
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.dto.LoginData;
+import com.codesoom.assignment.dto.SessionResponseData;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+
+/**
+ * ACCESS TOKEN을 발급하거나 검증하여 처리 결과를 반환한다.
+ */
+@RestController
+@RequestMapping("session")
+public class SessionController {
+    private final AuthenticationService authenticationService;
+
+    private final UserService userService;
+
+    public SessionController(AuthenticationService authenticationService,
+                             UserService userService) {
+        this.authenticationService = authenticationService;
+        this.userService = userService;
+    }
+
+    /**
+     * 로그인 데이터를 처리하고 회원이라면 토큰을 발급합니다.
+     * @param loginData 토큰을 발급받기 위한 로그인 데이터
+     * @return SessionResponseData 토큰은 담고있는 응답 객체
+     */
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public SessionResponseData login(@RequestBody @Valid LoginData loginData) {
+        User user = userService.findUserByEmailPassword(loginData);
```

Comment:
LoginData는 외부세계에서 컨트롤러라는 Layer로 데이터를 전송할 때 사용되는 객체입니다. 그래서 우리는 DTO라고 부릅니다.
DTO는 데이터를 전송할 때 받는 입장에서 사용하기 좋은 형식으로 전달해야 합니다.
따라서 여기서 LoginData를 그대로 전달하기 보다는 `email`과 `password`로 각각 전달하는게 더 좋을 것 같습니다.

```java
User user = userService.findUserByEmailPassword(email, password);
```

> Typically the data that crosses the boundaries is simple data structures. You can use basic structs or simple Data Transfer objects if you like. Or the data can simply be arguments in function calls. Or you can pack it into a hashmap, or construct it into an object. The important thing is that isolated, simple, data structures are passed across the boundaries. We don’t want to cheat and pass Entities or Database rows. We don’t want the data structures to have any kind of dependency that violates The Dependency Rule.

> For example, many database frameworks return a convenient data format in response to a query. We might call this a RowStructure. We don’t want to pass that row structure inwards across a boundary. That would violate The Dependency Rule because it would force an inner circle to know something about an outer circle.

> So when we pass data across a boundary, it is always in the form that is most convenient for the inner circle.

See also
  - https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html

Reply:
와우... 참고 자료 너무 감사합니다. _**clean code** 지향하겠습니다 !!_